### DISCUSSION TITLE
RT-dependent Normalization when samples have a few unique, high intensity proteins

### DISCUSSION THREAD

**Discussion by DavidMery** [vdemichev/DiaNN#1089] [General]
Vadim,

Thank you for your excellent contribution of DIA-NN to the proteomics community. 

BACKGROUND ISSUE
I have DIA data on purified plasma cells from myeloma patient samples. Myeloma plasma cells from each patient can have one of several translocations involving Chromosome 14 with another chromosome (e.g. 4, 11, 16, 20). This results in each subset of patients having generally the same plasma cell proteome with the exception of a few unique and highly intense spike proteins compared to other translocation types (e.g. t(4;14) vs t(14:16)).

I have paired samples with RNA (microarray) data showing that I should see these spikes in the protein, which I do, but I am also getting measurements for these spike protein in some patients that do not have these translocations. 

In issue #1062 you said " It happens quite often that specific runs have relative depletion of...peptides around a specific RT. RT-dependent normalisation corrects for it."

QUESTION
1. Would the presence of high intensity spike proteins in patient samples with a particular translocation effect the RT-dependent Normalization across all other samples by normalizing noise into a signal detectable by the MBR algorithm?

Thank you for your time and please feel free to correct an misunderstanding I may have,

**Comment by vdemichev** [comment]
Hi David,

No, I don't think this can affect normalisation.
However what you probably observe is misidentification.
What I can suggest here is filter the data by PG.PEP <= 0.01, i.e. this will give you high run-specific confidence of protein identifications.

Best,
Vadim

**Reply by DavidMery** [reply]
Hello Vadim,

First, thank you for your advice. I ran RT-dependent normalization and no normalization and the RT-dependent normalization really improves the data!

QUESTION:
I want to filter my peptides by PEP <0.01, but I want also want to have the PG.Quantity and PG.Normalized updated to reflect this change. Do I need to add the PEP <0.01 to the Additional Options window in the GUI as a command before I run the analysis in DIA-NN? If so, what would this code/command look like? 

Thank you,
David

**Reply by vdemichev** [reply]
Hi David,

> but I want also want to have the PG.Quantity and PG.Normalized updated to reflect this change

Then need to recalculate protein quantities using the iq R package (results will be different from what DIA-NN calculates internally though, as DIA-NN uses a modified version of MaxLFQ or QuantUMS).

However, this is unlikely to result in better protein quantities - it is usually preferable to use more peptides, even if those are less confident.

What makes total sense though - filtering based on PG.PEP. This is simple, just filter the main .parquet report, i.e. no need to recalculate quantities.

Best,
Vadim

**Comment by DavidMery** [comment]
Hello Vadim,

After filtering by PG.PEP <=0.01 in the main parquet file, I would like to recreate the PG.Matrix. My work requires me to do a lot in Excel. I looked into the code to generate a matrix (https://rdrr.io/github/vdemichev/diann-rpackage/man/diann_matrix.html)

QUESTION: Can I add the PG.PEP <= 0.01 criteria to this so that I can generate a matrix for Excel use?

Thank you,

[diann_matrix](https://rdrr.io/github/vdemichev/diann-rpackage/man/diann_matrix.html)(
  x,
  id.header = "Precursor.Id",
  quantity.header = "Precursor.Normalised",
  proteotypic.only = F,
  [q](https://rdrr.io/r/base/quit.html) = 0.01,
  protein.q = 1,
  pg.q = 1,
  gg.q = 1
)

**Reply by vdemichev** [reply]
Hi David,

I am not sure if diann R package actually works fine with .parquet files, I think I need to update it.
You can also just use the function:
```
library(reshape2)
library(dcast)
pivot <- function(df, file.header, id.header, quantity.header) {
  df <- unique(df[,c(file.header, id.header, quantity.header)])
  melted <- as.data.frame(melt(df, id.vars = c(file.header, id.header), measure.vars = c(quantity.header)))
  melted$value[which(melted$value == 0)] <- NA
  piv <- dcast(melted, as.formula(paste0(id.header,'~',file.header)), value.var = "value") 
  rownames(piv) <- piv[[1]]
  piv[,1] <- NULL
  as.matrix(piv)
}
```

In either case, diann_matrix() or pivot(), can just call it like this: pivot(x[x$PEP <= 0.01,],...)

Best,
Vadim
